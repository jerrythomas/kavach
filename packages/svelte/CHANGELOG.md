# @kavach/svelte

## 1.0.0-next.6

### Patch Changes

- Added store for auth status and used alerts to display status on demo page.
- Updated dependencies
  - @kavach/cookie@1.0.0-next.6
  - @kavach/core@1.0.0-next.6
  - @kavach/deflector@1.0.0-next.6
  - @kavach/logger@1.0.0-next.6

## 1.0.0-next.5

### Patch Changes

- chore: release 1.0.0-next.4
- Split core package into logger and deflector
- 1a35ad0: Resolved linter errors.
- Releasing the next version.
- Updated dependencies
- Updated dependencies
- Updated dependencies [1a35ad0]
- Updated dependencies
  - @kavach/cookie@1.0.0-next.5
  - @kavach/core@1.0.0-next.5
  - @kavach/deflector@1.0.0-next.5
  - @kavach/logger@1.0.0-next.5
