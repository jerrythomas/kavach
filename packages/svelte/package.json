{
	"name": "@kavach/svelte",
	"version": "1.0.0-next.2",
	"description": "Drop in security for web apps.",
	"author": "Jerry Thomas <me@jerrythomas.name>",
	"license": "MIT",
	"main": "index.js",
	"svelte": "src/index.js",
	"module": "src/index.js",
	"types": "dist/index.d.ts",
	"type": "module",
	"repository": {
		"type": "git",
		"url": "https://github.com/jerrythomas/kavach",
		"directory": "packages/kavach"
	},
	"prettier": "@jerrythomas/prettier-config",
	"publishConfig": {
		"access": "public"
	},
	"scripts": {
		"prepublishOnly": "tsc",
		"lint": "prettier --check --plugin-search-dir=. . && eslint .",
		"format": "prettier --write --plugin-search-dir=. .",
		"test:ci": "vitest run",
		"test:ui": "vitest --ui",
		"test": "vitest",
		"coverage": "vitest run --coverage",
		"upgrade": "pnpm upgrade"
	},
	"devDependencies": {
		"@jerrythomas/prettier-config": "^1.0.0",
		"@sveltejs/vite-plugin-svelte": "^2.0.2",
		"@testing-library/svelte": "^3.2.2",
		"@vitest/ui": "~0.12.10",
		"c8": "^7.12.0",
		"eslint-config-svelte": "workspace:*",
		"js-yaml": "^4.1.0",
		"jsdom": "^19.0.0",
		"shared-config": "workspace:^1.0.0",
		"svelte": "^3.55.0",
		"typescript": "^4.9.4",
		"vite": "^4.0.3",
		"vitest": "~0.19.1"
	},
	"exports": {
		"./package.json": "./package.json",
		".": {
			"types": "./types/index.d.ts",
			"import": "./src/index.js"
		}
	},
	"dependencies": {
		"@kavach/cookie": "workspace:1.0.0-next.2",
		"@kavach/core": "workspace:*",
		"@svelte-spice/form": "1.0.0-next.3",
		"@svelte-spice/input": "1.0.0-next.3",
		"ramda": "^0.28.0"
	}
}
