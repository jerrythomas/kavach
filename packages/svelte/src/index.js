export * from './kavach'
export * from './request'
export * from './endpoints'

export { default as AuthButton } from './AuthButton.svelte'
export { default as AuthProvider } from './AuthProvider.svelte'
export { default as AuthGroup } from './AuthGroup.svelte'
export { default as AuthError } from './AuthError.svelte'
export { default as AuthResponse } from './AuthResponse.svelte'
export { default as AuthHandler } from './AuthHandler.svelte'
export { default as Logout } from './Logout.svelte'
