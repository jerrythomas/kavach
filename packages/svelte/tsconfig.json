{
  "extends": "shared-config/jsconfig.json",
  "include": ["src/**/*.js", "src/**/*.svelte"],
  "compilerOptions": {
    "outDir": "dist"
  }
}
