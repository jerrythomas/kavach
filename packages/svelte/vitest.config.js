import { defineConfig } from 'vite'
import { svelte } from '@sveltejs/vite-plugin-svelte'
import { testConfig } from 'shared-config'

testConfig.coverage.provider = 'v8'
export default defineConfig({
	plugins: [svelte({ hot: !process.env.VITEST })],
	test: testConfig
})
