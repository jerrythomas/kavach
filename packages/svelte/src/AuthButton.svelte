<script>
	import { IconButton } from '@svelte-spice/form'
	import { getContext } from 'svelte'

	const kavach = getContext('kavach')
	// export let kavach
	/** @type {string} */
	export let provider
	/** @type {string} */
	export let label
	/** @type {Array<string>} */
	export let scopes = []
	/** @type {Array<string>} */
	export let params = []

	function signIn() {
		// console.log('kavach', kavach)
		kavach.adapter.auth.signInWithOAuth({
			provider,
			options: { scopes, params, redirectTo: kavach.page.auth }
		})
	}
</script>

<!-- <input type="hidden" name="provider" value={provider} />
<input type="hidden" name="scopes" value={JSON.stringify(scopes)} />
<input type="hidden" name="params" value={JSON.stringify(params)} /> -->
<IconButton on:click={signIn} {label} leftIcon="logo-{provider}" />
