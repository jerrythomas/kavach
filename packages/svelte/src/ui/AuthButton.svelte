<script>
	import { Button } from '@rokkit/molecules'
	import { createEventDispatcher } from 'svelte'

	const dispatch = createEventDispatcher()

	/** @type {string} */
	export let provider
	/** @type {string} */
	export let label
	/** @type {Array<string>} */
	export let scopes = []

	function handle() {
		dispatch('click', { provider, scopes })
	}
</script>

<Button on:click={handle} {label} leftIcon="logo-{provider}" />
