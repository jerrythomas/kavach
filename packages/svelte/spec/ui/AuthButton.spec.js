import { describe, expect, it, beforeEach, vi } from 'vitest'
import { cleanup, render, fireEvent } from '@testing-library/svelte'
import { toHaveBeenDispatchedWith } from './validators'
import AuthButton from '../../src/ui/AuthButton.svelte'

expect.extend({ toHaveBeenDispatchedWith })

describe('AuthButton.svelte', () => {
	const handleClick = vi.fn()

	beforeEach(() => {
		cleanup()
	})

	it('should render using provider', async () => {
		const { container, component } = render(AuthButton, {
			props: { provider: 'google', label: 'Sign in with Google' }
		})
		const button = container.querySelector('button')

		component.$on('click', handleClick)
		expect(container).toMatchSnapshot()

		await fireEvent.click(button)
		expect(handleClick).toHaveBeenDispatchedWith({
			provider: 'google',
			scopes: []
		})
	})

	// it('should render using label', async () => {
	// 	const { container, component } = render(AuthButton, {
	// 		props: { provider: 'google', label: 'Sign in with Google' }
	// 	})
	// 	const button = container.querySelector('button')

	// 	component.$on('click', handleClick)
	// 	expect(container).toMatchSnapshot()

	// 	await fireEvent.click(button)
	// 	expect(handleClick).toHaveBeenDispatchedWith({
	// 		provider: 'google',
	// 		scopes: []
	// 	})
	// })

	it('should include scopes', async () => {
		const { container, component } = render(AuthButton, {
			props: {
				provider: 'azure',
				label: 'Sign in with Azure',
				scopes: ['email']
			}
		})
		const button = container.querySelector('button')

		component.$on('click', handleClick)
		expect(container).toMatchSnapshot()

		await fireEvent.click(button)
		expect(handleClick).toHaveBeenDispatchedWith({
			provider: 'azure',
			scopes: ['email']
		})
	})
})
