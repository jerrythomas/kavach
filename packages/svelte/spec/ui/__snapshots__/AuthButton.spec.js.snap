// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthButton.svelte > should include scopes 1`] = `
<div>
  <button
    class="flex flex-row items-center "
    type="submit"
  >
    <icon-left
      class="flex flex-row"
    >
      <icon
        aria-label="logo-azure"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="logo-azure"
        />
      </icon>
    </icon-left>
     
    <p>
      Sign in with Azure
    </p>
    
     
  </button>
</div>
`;

exports[`AuthButton.svelte > should render using label 1`] = `
<div>
  <button
    class="flex flex-row items-center "
    type="submit"
  >
    <icon-left
      class="flex flex-row"
    >
      <icon
        aria-label="logo-google"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="logo-google"
        />
      </icon>
    </icon-left>
     
    <p>
      Sign in with Google
    </p>
    
     
  </button>
</div>
`;

exports[`AuthButton.svelte > should render using provider 1`] = `
<div>
  <button
    class="flex flex-row items-center "
    type="submit"
  >
    <icon-left
      class="flex flex-row"
    >
      <icon
        aria-label="logo-google"
        class="flex flex-shrink-0 items-center justify-center "
        role="img"
        tabindex="-1"
      >
        <i
          aria-hidden="true"
          class="logo-google"
        />
      </icon>
    </icon-left>
     
    
     
  </button>
</div>
`;
